tests/users/run.sh

```diff
@@ -0,0 +1,282 @@
+done
+for ((i = 4096; i < $((4096+${numberOfUsersToTest}+1)); ++i)); do
+_RunSql "create user user_test_8_${i}
+         with REAL_NAME = 'TestUser ${i} 😀', DESCRIPTION = 'User for developer ${i} 😀'"
+done
+for ((i = 4096; i < $((4096+${numberOfUsersToTest}+1)); ++i)); do
+_RunSql "create user user_test_9_${i}
+         with STATE = ACTIVE, REAL_NAME = 'TestUser ${i} 😀', DESCRIPTION = 'User for developer ${i} 😀'"
+done
+#### Create max users
+for ((i = $((4096+${numberOfUsersToTest}+1)); i < $(($((4096+${numberOfUsersToTest}+1))+${numberOfUsersToTestMax}+1)); ++i)); do
+_RunSql "create user user_test_10_${i}
+         with STATE = ACTIVE, REAL_NAME = 'TestUser ${i} 😀', DESCRIPTION = 'User for developer ${i} 😀'"
+done
+#### Alter user
+for ((i = 4096; i < $((4096+${numberOfUsersToTest}+1)); ++i)); do
+_RunSql "alter user user_test_1_${i}
+         set STATE = ACTIVE"
+_RunSqlAndValidateOutput "select STATE from SYS.SYS_USERS where name = 'USER_TEST_1_${i}'" "1"
+done
+for ((i = 4096; i < $((4096+${numberOfUsersToTest}+1)); ++i)); do
+_RunSql "alter user user_test_1_${i}
+         set STATE = INACTIVE"
+_RunSqlAndValidateOutput "select STATE from SYS.SYS_USERS where name = 'USER_TEST_1_${i}'" "0"
+done
+for ((i = 4096; i < $((4096+${numberOfUsersToTest}+1)); ++i)); do
+_RunSql "alter user user_test_1_${i}
+         set STATE = ACTIVE, REAL_NAME = 'TestUser updated 1 ${i} 😀'"
+_RunSqlAndValidateOutput "select STATE from SYS.SYS_USERS where name = 'USER_TEST_1_${i}'" "1"
+_RunSqlAndValidateOutput "select REAL_NAME from SYS.SYS_USERS where name = 'USER_TEST_1_${i}'" "TestUser updated 1 ${i} 😀"
+done
+for ((i = 4096; i < $((4096+${numberOfUsersToTest}+1)); ++i)); do
+_RunSql "alter user user_test_1_${i}
+         set STATE = ACTIVE, REAL_NAME = 'TestUser updated 2 ${i} 😀', DESCRIPTION = 'User for developer updated 2 ${i} 😀'"
+_RunSqlAndValidateOutput "select STATE from SYS.SYS_USERS where name = 'USER_TEST_1_${i}'" "1"
+_RunSqlAndValidateOutput "select REAL_NAME from SYS.SYS_USERS where name = 'USER_TEST_1_${i}'" "TestUser updated 2 ${i} 😀"
+_RunSqlAndValidateOutput "select DESCRIPTION from SYS.SYS_USERS where name = 'USER_TEST_1_${i}'" "User for developer updated 2 ${i} 😀"
+done
+for ((i = 4096; i < $((4096+${numberOfUsersToTest}+1)); ++i)); do
+_RunSql "alter user user_test_1_${i}
+         set REAL_NAME = 'TestUser updated 3 ${i} 😀', DESCRIPTION = 'User for developer updated 3 ${i} 😀'"
+_RunSqlAndValidateOutput "select REAL_NAME from SYS.SYS_USERS where name = 'USER_TEST_1_${i}'" "TestUser updated 3 ${i} 😀"
+_RunSqlAndValidateOutput "select DESCRIPTION from SYS.SYS_USERS where name = 'USER_TEST_1_${i}'" "User for developer updated 3 ${i} 😀"
+done
+
+#### Create user with wrong chars
+_RunSqlAndValidateOutput "create user 😀😀😀😀😀😀😀" 'Status 2: at .*: mismatched input'
+_CheckLogFiles 'common parse error'
+_RunSqlAndValidateOutput "create user 123456789" 'Status 2: at .*: mismatched input'
+_CheckLogFiles 'common parse error'
```

Comment:
note that if test for error message for the previous SQL (with user name with non alphabetic characters) passes and then this SQL (with numeric user name) for some reason works, the log check will still pass because it will find log occurrence made by execution of the previous statement. I think this can be somehow distinguished using timestamps, i.e not only message must match but also timestamp must be greater than timestamp picked right before running sql  

applies to all similar cases

Reply:
@cxxman actually the $1 parameter of `_CheckLogFiles` is a pattern to skip in the check and not a check that must be done.
 `_CheckLogFiles ` checks for `error` that are different from $1. If it finds one, then an unexpected error occurred and it exists 1

Reply:
And that different error may come from some previous test which was ok to generate that different error. So even with this approach we need to know from starting which point we should be looking for a "different error".